webapp/channels/src/components/file_preview_modal/file_preview_modal.tsx

```diff
@@ -306,7 +332,8 @@ export default class FilePreviewModal extends React.PureComponent<Props, State>
 
         if (!isFileInfo(fileInfo) || !fileInfo.archived) {
             if (this.state.loaded[this.state.imageIndex]) {
-                if (fileType === FileTypes.IMAGE || fileType === FileTypes.SVG) {
+                if (fileType === FileTypes.IMAGE || fileType === FileTypes.SVG ||
+                    (!isFileInfo(fileInfo) && this.isImageUrl(fileInfo.link))) {
```

Comment:
If `Utils.getFileType` now handles non-file-info, do we need to check this again?

Reply:
very valid point and also, since in this last refactor I've enhanced Utils.getFileType to properly handle LinkInfo objects by passing the full URL, the additional check is indeed redundant.